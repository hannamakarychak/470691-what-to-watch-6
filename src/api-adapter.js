export const adaptMovieToClient = (movie) => ({
  id: movie.id,
  name: movie.name,
  posterImage: movie.poster_image,
  previewImage: movie.preview_image,
  backgroundImage: movie.background_image,
  backgroundColor: movie.background_color,
  videoLink: movie.video_link,
  previewVideoLink: movie.preview_video_link,
  description: movie.description,
  rating: movie.rating,
  scoresCount: movie.scores_count,
  director: movie.director,
  starring: movie.starring,
  runTime: movie.run_time,
  genre: movie.genre,
  released: movie.released,
  isFavorite: movie.is_favorite,
});

export const adaptUserToClient = (user) => ({
  id: user.id,
  email: user.email,
  name: user.name,
  avatarUrl: user.avatar_url,
});
