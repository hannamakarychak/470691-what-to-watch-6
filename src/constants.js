export const ALL_GENRES = `All genres`;

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`
};
