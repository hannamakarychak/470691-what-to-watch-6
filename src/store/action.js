export const ActionType = {
  SET_GENRE: `main/setGenre`,
  GET_MOVIES_BY_GENRE: `main/getMoviesByGenre`
};

export const ActionCreator = {
  setGenre: (genre) => ({
    type: ActionType.SET_GENRE,
    payload: genre
  }),

  getListByGenre: () => ({
    type: ActionType.GET_MOVIES_BY_GENRE,
    payload: []
  })
};
