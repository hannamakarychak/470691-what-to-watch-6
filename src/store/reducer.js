import {ALL_GENRES, AuthorizationStatus} from "../constants";
import {ActionType} from "./action";

const initialState = {
  genre: ALL_GENRES,
  list: [],
  isDataLoaded: false,
  authorizationStatus: AuthorizationStatus.NO_AUTH
};

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.SET_GENRE:
      return {
        ...state,
        genre: action.payload
      };

    case ActionType.GET_ALL_MOVIES:
      return {
        ...state,
        isDataLoaded: true,
        list: action.payload
      };

    case ActionType.REQUIRED_AUTHORIZATION:
      return {
        ...state,
        authorizationStatus: action.payload
      };

    default:
      return state;
  }
};
