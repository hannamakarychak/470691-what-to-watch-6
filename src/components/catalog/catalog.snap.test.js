import React from 'react';
import {render} from '@testing-library/react';
import configureStore from 'redux-mock-store';
import * as redux from 'react-redux';
import {createMemoryHistory} from 'history';
import {Router} from 'react-router';

import Catalog from '../catalog/catalog';
import {NameSpace} from '../../store/reducer';
import {ALL_GENRES} from '../../constants';
import {mockMovie} from '../../store/action.test';
import {adaptMovieToClient} from '../../api-adapter';

jest.mock(`../player/player.jsx`, () => () => `PlayerComponent`);

const mockStore = configureStore({});

describe(`Should Catalog render correctly`, () => {
  it(`without load more button`, () => {
    const history = createMemoryHistory();
    const store = mockStore({
      [NameSpace.ALL_MOVIES]: {
        list: [adaptMovieToClient(mockMovie)],
        genre: ALL_GENRES
      },
    });
    const {container} = render(
      <redux.Provider store={store}>
        <Router history={history}>
          <Catalog />
        </Router>
      </redux.Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it(`with load more button`, () => {
    const history = createMemoryHistory();
    const store = mockStore({
      [NameSpace.ALL_MOVIES]: {
        list: new Array(9).fill().map((movie, i) => ({
          ...adaptMovieToClient(mockMovie),
          id: i
        })),
        genre: ALL_GENRES
      },
    });
    const {container} = render(
      <redux.Provider store={store}>
        <Router history={history}>
          <Catalog />
        </Router>
      </redux.Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
