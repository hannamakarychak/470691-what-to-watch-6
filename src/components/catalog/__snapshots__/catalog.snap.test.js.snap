// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Catalog render correctly with load more button 1`] = `
<div>
  <section
    class="catalog"
  >
    <h2
      class="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <ul
      class="catalog__genres-list"
    >
      <li
        class="catalog__genres-item catalog__genres-item--active"
      >
        <a
          class="catalog__genres-link"
          href="#"
        >
          All genres
        </a>
      </li>
      <li
        class="catalog__genres-item "
      >
        <a
          class="catalog__genres-link"
          href="#"
        >
          Drama
        </a>
      </li>
    </ul>
    <div
      class="catalog__movies-list"
    >
      <article
        class="small-movie-card catalog__movies-card"
      >
        <a
          class="small-movie-card__link"
          href="/films/0"
        >
          <div
            class="small-movie-card__image"
          >
            PlayerComponent
          </div>
          <h3
            class="small-movie-card__title"
          >
            Macbeth
          </h3>
        </a>
      </article>
      <article
        class="small-movie-card catalog__movies-card"
      >
        <a
          class="small-movie-card__link"
          href="/films/1"
        >
          <div
            class="small-movie-card__image"
          >
            PlayerComponent
          </div>
          <h3
            class="small-movie-card__title"
          >
            Macbeth
          </h3>
        </a>
      </article>
      <article
        class="small-movie-card catalog__movies-card"
      >
        <a
          class="small-movie-card__link"
          href="/films/2"
        >
          <div
            class="small-movie-card__image"
          >
            PlayerComponent
          </div>
          <h3
            class="small-movie-card__title"
          >
            Macbeth
          </h3>
        </a>
      </article>
      <article
        class="small-movie-card catalog__movies-card"
      >
        <a
          class="small-movie-card__link"
          href="/films/3"
        >
          <div
            class="small-movie-card__image"
          >
            PlayerComponent
          </div>
          <h3
            class="small-movie-card__title"
          >
            Macbeth
          </h3>
        </a>
      </article>
      <article
        class="small-movie-card catalog__movies-card"
      >
        <a
          class="small-movie-card__link"
          href="/films/4"
        >
          <div
            class="small-movie-card__image"
          >
            PlayerComponent
          </div>
          <h3
            class="small-movie-card__title"
          >
            Macbeth
          </h3>
        </a>
      </article>
      <article
        class="small-movie-card catalog__movies-card"
      >
        <a
          class="small-movie-card__link"
          href="/films/5"
        >
          <div
            class="small-movie-card__image"
          >
            PlayerComponent
          </div>
          <h3
            class="small-movie-card__title"
          >
            Macbeth
          </h3>
        </a>
      </article>
      <article
        class="small-movie-card catalog__movies-card"
      >
        <a
          class="small-movie-card__link"
          href="/films/6"
        >
          <div
            class="small-movie-card__image"
          >
            PlayerComponent
          </div>
          <h3
            class="small-movie-card__title"
          >
            Macbeth
          </h3>
        </a>
      </article>
      <article
        class="small-movie-card catalog__movies-card"
      >
        <a
          class="small-movie-card__link"
          href="/films/7"
        >
          <div
            class="small-movie-card__image"
          >
            PlayerComponent
          </div>
          <h3
            class="small-movie-card__title"
          >
            Macbeth
          </h3>
        </a>
      </article>
    </div>
    <div
      class="catalog__more"
    >
      <button
        class="catalog__button"
        type="button"
      >
        Show more
      </button>
    </div>
  </section>
</div>
`;

exports[`Should Catalog render correctly without load more button 1`] = `
<div>
  <section
    class="catalog"
  >
    <h2
      class="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <ul
      class="catalog__genres-list"
    >
      <li
        class="catalog__genres-item catalog__genres-item--active"
      >
        <a
          class="catalog__genres-link"
          href="#"
        >
          All genres
        </a>
      </li>
      <li
        class="catalog__genres-item "
      >
        <a
          class="catalog__genres-link"
          href="#"
        >
          Drama
        </a>
      </li>
    </ul>
    <div
      class="catalog__movies-list"
    >
      <article
        class="small-movie-card catalog__movies-card"
      >
        <a
          class="small-movie-card__link"
          href="/films/1"
        >
          <div
            class="small-movie-card__image"
          >
            PlayerComponent
          </div>
          <h3
            class="small-movie-card__title"
          >
            Macbeth
          </h3>
        </a>
      </article>
    </div>
  </section>
</div>
`;
